### DISCUSSION TITLE
PC controlling Android: keyboard support?

### DISCUSSION THREAD

**Discussion by mykiwi** [rustdesk/rustdesk#681] [Q&A] [answered]
Hi,

I successfully connect my Android device on my PC and I can navigate in it using my mouse 😍 

But I can not use my keyboard PC to write in Android...
Do you know if this would be possible in a near future? Or did I miss something?

Thanks

**Comment by Heap-Hop** [comment, marked as answer]
AFAIK, injecting keyboard events to Android without `root` or `adb` is impossible. Android hasn't given us such a permission yet (If I'm wrong, please tell me).
We want to make a portable and safe mobile APP, so we don't plan to use `root` or `abd` in a near future.

[knox.remotecontrol](https://docs.samsungknox.com/devref/knox-sdk/reference/com/samsung/android/knox/remotecontrol/package-summary.html)
`Knox` another way in Samsung Android, which supports keyboard events. It's in our plan.

**Reply by mykiwi** [reply]
Thank you for the explanation

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/3000
